src/app/scouter/page.tsx

```diff
@@ -0,0 +1,96 @@
+'use client';
+
+import { useRouter } from 'next/navigation';
+import { ScouterViewer } from '@/components/ui/ScouterViewer';
+import { Button, createButtonProps } from '@/components/ui/Button';
+import { useState, useEffect } from 'react';
+import { getBattlePower } from '@/app/calc/ScouterCalculator';
+
+// 戦闘力計算用コンポーネント
+function ResultButton({ follower }: { follower: number }) {
+  const router = useRouter();
+  
+  const handleSubmit = () => {
+    const battlePower = getBattlePower(follower);
+    try {
+      sessionStorage.setItem('battlePower',  String(battlePower));
+      router.push('/result');
+    } catch (e) {
+      console.error('セッションストレージへの保存に失敗:', e);
+      // ユーザーにエラーを通知する処理などをここに記述
+    }
+  };
+  
+  return (
+    <div className="mt-2">
+      <Button
+        button={createButtonProps('button', '結果を見る')}
+        onClick={handleSubmit}
+      />
+    </div>
+  );
+}
+
+// ScouterViewerを表示するコンポーネント
+function ScouterSection({ onRotationComplete }: { onRotationComplete: () => void }) {
+  return (
+    <ScouterViewer onRotationComplete={onRotationComplete} />
+  );
+}
+
+export default function ScouterPage() {
+  const router = useRouter();
+  const [followerNumber, setFollowerNumber] = useState<number | null>(null);
+  const [rotationCompleted, setRotationCompleted] = useState(false);
+  const [isRedirecting, setIsRedirecting] = useState(false);
+
+  // セッションストレージからフォロワー数を取得
+  useEffect(() => {
+    const storedFollower = sessionStorage.getItem('follower');
+    if (!storedFollower) {
+      setIsRedirecting(true);
+      return;
+    }
+
+    const numFollower = Number(storedFollower);
+    if (isNaN(numFollower)) {
+      setIsRedirecting(true);
+      return;
+    }
+
+    setFollowerNumber(numFollower);
+  }, []);
+
+  // リダイレクト処理
+  useEffect(() => {
+    if (isRedirecting) {
+      router.push('/');
+    }
+  }, [isRedirecting, router]);
```

Comment:
ここでセッションストレージの値を削除するのもシンプルでいいかもです！
画面遷移時なので、これ以降はもうこのセッションストレージは使わないよ。という意味を明示もできますね
```suggestion
  // リダイレクト処理
  useEffect(() => {
    if (isRedirecting) {
      router.push('/');
      sessionStorage.removeItem('follower');
    }
  }, [isRedirecting, router]);
```